File: mobile/colors/StormSkyColors.swift
//
//  StormSkyColors.swift
//  ALADDIN Mobile App
//
//  Unified Color Scheme - "Грозовое небо с золотыми акцентами"
//  Для iOS приложения ALADDIN
//
//  Created: 2025-01-27
//  Version: 2.0.0
//

import SwiftUI

/// Единая цветовая схема "Грозовое небо" для всего приложения ALADDIN
struct StormSkyColors {
    
    // MARK: - Storm Sky Colors (Грозовое небо)
    
    /// Темно-синий глубокий - верх и низ неба
    static let stormSkyDark = Color(hex: "#0a1128")
    
    /// Синий грозового неба - основной цвет
    static let stormSkyMain = Color(hex: "#1e3a5f")
    
    /// Средний синий - центр неба
    static let stormSkyMid = Color(hex: "#2e5090")
    
    // MARK: - Gold Accents (Золотые акценты)
    
    /// Золотой основной - заголовки, кнопки, акценты
    static let goldMain = Color(hex: "#F59E0B")
    
    /// Золотой светлый - hover эффекты
    static let goldLight = Color(hex: "#FCD34D")
    
    /// Золотой темный - тени, градиенты
    static let goldDark = Color(hex: "#D97706")
    
    // MARK: - Status Colors (Цвета статусов)
    
    /// Изумрудный успех - статус "Подключено"
    static let successGreen = Color(hex: "#10B981")
    
    /// Рубиновый ошибка - статус "Отключено"
    static let errorRed = Color(hex: "#EF4444")
    
    /// Голубой молнии - эффекты, анимации
    static let lightningBlue = Color(hex: "#60A5FA")
    
    // MARK: - Text Colors (Цвета текста)
    
    /// Белый чистый - основной текст
    static let white = Color.white
    
    /// Белый полупрозрачный - вторичный текст
    static let whiteSecondary = Color.white.opacity(0.8)
    
    /// Темно-серый - текст на светлом фоне
    static let darkGray = Color(hex: "#1F2937")
    
    // MARK: - Background Colors (Цвета фона)
    
    /// Основной фон - грозовое небо
    static let backgroundPrimary = stormSkyMain
    
    /// Карточки - белый с прозрачностью
    static let cardBackground = Color.white.opacity(0.95)
    
    /// Вторичный фон
    static let backgroundSecondary = stormSkyDark
    
    // MARK: - Gradients (Градиенты)
    
    /// Градиент грозового неба
    static let stormSkyGradient = LinearGradient(
        gradient: Gradient(colors: [stormSkyDark, stormSkyMain, stormSkyMid, stormSkyMain, stormSkyDark]),
        startPoint: .top,
        endPoint: .bottom
    )
    
    /// Золотой градиент
    static let goldGradient = LinearGradient(
        gradient: Gradient(colors: [goldMain, goldLight, goldDark]),
        startPoint: .leading,
        endPoint: .trailing
    )
    
    /// Градиент успеха
    static let successGradient = LinearGradient(
        gradient: Gradient(colors: [successGreen, Color(hex: "#059669")]),
        startPoint: .leading,
        endPoint: .trailing
    )
    
    /// Градиент ошибки
    static let errorGradient = LinearGradient(
        gradient: Gradient(colors: [errorRed, Color(hex: "#DC2626")]),
        startPoint: .leading,
        endPoint: .trailing
    )
    
    // MARK: - Shadows (Тени)
    
    /// Золотая тень
    static let goldShadow = Color.black.opacity(0.3)
    
    /// Синяя тень
    static let blueShadow = lightningBlue.opacity(0.3)
    
    /// Зеленая тень
    static let greenShadow = successGreen.opacity(0.3)
    
    /// Красная тень
    static let redShadow = errorRed.opacity(0.3)
}

// MARK: - Color Extensions

extension Color {
    /// Инициализация цвета из HEX строки
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3: // RGB (12-bit)
            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6: // RGB (24-bit)
            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8: // ARGB (32-bit)
            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default:
            (a, r, g, b) = (1, 1, 1, 0)
        }
        
        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue:  Double(b) / 255,
            opacity: Double(a) / 255
        )
    }
}

// MARK: - View Modifiers

extension View {
    /// Применение золотого акцента
    func goldAccent() -> some View {
        self.foregroundColor(StormSkyColors.goldMain)
            .shadow(color: StormSkyColors.goldShadow, radius: 2, x: 0, y: 1)
    }
    
    /// Применение грозового неба фона
    func stormSkyBackground() -> some View {
        self.background(StormSkyColors.stormSkyGradient)
    }
    
    /// Применение карточки в стиле грозового неба
    func stormCard() -> some View {
        self.background(StormSkyColors.cardBackground)
            .overlay(
                RoundedRectangle(cornerRadius: 20)
                    .stroke(StormSkyColors.goldMain.opacity(0.3), lineWidth: 2)
            )
            .cornerRadius(20)
            .shadow(color: StormSkyColors.goldShadow, radius: 10, x: 0, y: 5)
    }
    
    /// Применение золотой кнопки
    func goldButton() -> some View {
        self.background(StormSkyColors.goldGradient)
            .foregroundColor(.white)
            .cornerRadius(25)
            .shadow(color: StormSkyColors.goldShadow, radius: 5, x: 0, y: 3)
    }
}

// MARK: - Previews

#if DEBUG
struct StormSkyColors_Previews: PreviewProvider {
    static var previews: some View {
        VStack(spacing: 20) {
            // Storm Sky Gradient
            Rectangle()
                .fill(StormSkyColors.stormSkyGradient)
                .frame(height: 100)
                .overlay(
                    Text("Грозовое небо")
                        .foregroundColor(.white)
                        .font(.title)
                        .bold()
                )
            
            // Gold Gradient
            Rectangle()
                .fill(StormSkyColors.goldGradient)
                .frame(height: 60)
                .overlay(
                    Text("Золотые акценты")
                        .foregroundColor(.white)
                        .font(.headline)
                        .bold()
                )
            
            // Color Palette
            HStack(spacing: 10) {
                Circle()
                    .fill(StormSkyColors.stormSkyDark)
                    .frame(width: 40, height: 40)
                
                Circle()
                    .fill(StormSkyColors.stormSkyMain)
                    .frame(width: 40, height: 40)
                
                Circle()
                    .fill(StormSkyColors.stormSkyMid)
                    .frame(width: 40, height: 40)
                
                Circle()
                    .fill(StormSkyColors.goldMain)
                    .frame(width: 40, height: 40)
                
                Circle()
                    .fill(StormSkyColors.successGreen)
                    .frame(width: 40, height: 40)
                
                Circle()
                    .fill(StormSkyColors.errorRed)
                    .frame(width: 40, height: 40)
            }
            
            // Storm Card Example
            VStack {
                Text("Пример карточки")
                    .font(.headline)
                    .goldAccent()
                
                Text("В стиле грозового неба")
                    .foregroundColor(StormSkyColors.darkGray)
            }
            .padding()
            .stormCard()
            
            // Gold Button Example
            Button("Золотая кнопка") {
                // Action
            }
            .padding()
            .goldButton()
        }
        .padding()
        .stormSkyBackground()
    }
}
#endif

File: mobile/colors/StormSkyColors.kt
package com.aladdin.mobile.colors

import android.graphics.Color
import androidx.compose.ui.graphics.Color as ComposeColor

/**
 * Unified Color Scheme - "Грозовое небо с золотыми акцентами"
 * Для Android приложения ALADDIN
 *
 * Created: 2025-01-27
 * Version: 2.0.0
 */
object StormSkyColors {
    
    // MARK: - Storm Sky Colors (Грозовое небо)
    
    /**
     * Темно-синий глубокий - верх и низ неба
     */
    val stormSkyDark = Color.parseColor("#0a1128")
    val stormSkyDarkCompose = ComposeColor(0x0a1128)
    
    /**
     * Синий грозового неба - основной цвет
     */
    val stormSkyMain = Color.parseColor("#1e3a5f")
    val stormSkyMainCompose = ComposeColor(0x1e3a5f)
    
    /**
     * Средний синий - центр неба
     */
    val stormSkyMid = Color.parseColor("#2e5090")
    val stormSkyMidCompose = ComposeColor(0x2e5090)
    
    // MARK: - Gold Accents (Золотые акценты)
    
    /**
     * Золотой основной - заголовки, кнопки, акценты
     */
    val goldMain = Color.parseColor("#F59E0B")
    val goldMainCompose = ComposeColor(0xF59E0B)
    
    /**
     * Золотой светлый - hover эффекты
     */
    val goldLight = Color.parseColor("#FCD34D")
    val goldLightCompose = ComposeColor(0xFCD34D)
    
    /**
     * Золотой темный - тени, градиенты
     */
    val goldDark = Color.parseColor("#D97706")
    val goldDarkCompose = ComposeColor(0xD97706)
    
    // MARK: - Status Colors (Цвета статусов)
    
    /**
     * Изумрудный успех - статус "Подключено"
     */
    val successGreen = Color.parseColor("#10B981")
    val successGreenCompose = ComposeColor(0x10B981)
    
    /**
     * Рубиновый ошибка - статус "Отключено"
     */
    val errorRed = Color.parseColor("#EF4444")
    val errorRedCompose = ComposeColor(0xEF4444)
    
    /**
     * Голубой молнии - эффекты, анимации
     */
    val lightningBlue = Color.parseColor("#60A5FA")
    val lightningBlueCompose = ComposeColor(0x60A5FA)
    
    // MARK: - Text Colors (Цвета текста)
    
    /**
     * Белый чистый - основной текст
     */
    val white = Color.WHITE
    val whiteCompose = ComposeColor.White
    
    /**
     * Белый полупрозрачный - вторичный текст
     */
    val whiteSecondary = Color.argb(204, 255, 255, 255) // 80% opacity
    val whiteSecondaryCompose = ComposeColor.White.copy(alpha = 0.8f)
    
    /**
     * Темно-серый - текст на светлом фоне
     */
    val darkGray = Color.parseColor("#1F2937")
    val darkGrayCompose = ComposeColor(0x1F2937)
    
    // MARK: - Background Colors (Цвета фона)
    
    /**
     * Основной фон - грозовое небо
     */
    val backgroundPrimary = stormSkyMain
    val backgroundPrimaryCompose = stormSkyMainCompose
    
    /**
     * Карточки - белый с прозрачностью
     */
    val cardBackground = Color.argb(242, 255, 255, 255) // 95% opacity
    val cardBackgroundCompose = ComposeColor.White.copy(alpha = 0.95f)
    
    /**
     * Вторичный фон
     */
    val backgroundSecondary = stormSkyDark
    val backgroundSecondaryCompose = stormSkyDarkCompose
    
    // MARK: - Gradient Colors (Цвета градиентов)
    
    /**
     * Цвета для градиента грозового неба
     */
    val stormSkyGradientColors = listOf(
        stormSkyDarkCompose,
        stormSkyMainCompose,
        stormSkyMidCompose,
        stormSkyMainCompose,
        stormSkyDarkCompose
    )
    
    /**
     * Цвета для золотого градиента
     */
    val goldGradientColors = listOf(
        goldMainCompose,
        goldLightCompose,
        goldDarkCompose
    )
    
    /**
     * Цвета для градиента успеха
     */
    val successGradientColors = listOf(
        successGreenCompose,
        ComposeColor(0x059669)
    )
    
    /**
     * Цвета для градиента ошибки
     */
    val errorGradientColors = listOf(
        errorRedCompose,
        ComposeColor(0xDC2626)
    )
    
    // MARK: - Shadow Colors (Цвета теней)
    
    /**
     * Золотая тень
     */
    val goldShadow = Color.argb(77, 0, 0, 0) // 30% opacity
    val goldShadowCompose = ComposeColor.Black.copy(alpha = 0.3f)
    
    /**
     * Синяя тень
     */
    val blueShadow = Color.argb(77, 96, 165, 250) // 30% opacity
    val blueShadowCompose = lightningBlueCompose.copy(alpha = 0.3f)
    
    /**
     * Зеленая тень
     */
    val greenShadow = Color.argb(77, 16, 185, 129) // 30% opacity
    val greenShadowCompose = successGreenCompose.copy(alpha = 0.3f)
    
    /**
     * Красная тень
     */
    val redShadow = Color.argb(77, 239, 68, 68) // 30% opacity
    val redShadowCompose = errorRedCompose.copy(alpha = 0.3f)
}

/**
 * Расширения для работы с цветами
 */
object ColorUtils {
    
    /**
     * Создание цвета из HEX строки
     */
    fun fromHex(hex: String): Color {
        return Color.parseColor(hex)
    }
    
    /**
     * Создание Compose цвета из HEX строки
     */
    fun fromHexCompose(hex: String): ComposeColor {
        val cleanHex = hex.replace("#", "")
        return ComposeColor(cleanHex.toLong(16))
    }
    
    /**
     * Получение контрастного цвета для текста
     */
    fun getContrastText(backgroundColor: Color): Color {
        val r = Color.red(backgroundColor)
        val g = Color.green(backgroundColor)
        val b = Color.blue(backgroundColor)
        
        // Расчет яркости
        val brightness = (r * 299 + g * 587 + b * 114) / 1000
        
        return if (brightness < 128) Color.WHITE else Color.BLACK
    }
    
    /**
     * Получение контрастного Compose цвета для текста
     */
    fun getContrastTextCompose(backgroundColor: ComposeColor): ComposeColor {
        val r = backgroundColor.red
        val g = backgroundColor.green
        val b = backgroundColor.blue
        
        // Расчет яркости
        val brightness = (r * 299 + g * 587 + b * 114) / 1000
        
        return if (brightness < 0.5f) ComposeColor.White else ComposeColor.Black
    }
}

/**
 * Константы для использования в XML ресурсах
 */
object StormSkyColorsRes {
    
    // Storm Sky Colors
    const val STORM_SKY_DARK = "#0a1128"
    const val STORM_SKY_MAIN = "#1e3a5f"
    const val STORM_SKY_MID = "#2e5090"
    
    // Gold Accents
    const val GOLD_MAIN = "#F59E0B"
    const val GOLD_LIGHT = "#FCD34D"
    const val GOLD_DARK = "#D97706"
    
    // Status Colors
    const val SUCCESS_GREEN = "#10B981"
    const val ERROR_RED = "#EF4444"
    const val LIGHTNING_BLUE = "#60A5FA"
    
    // Text Colors
    const val WHITE = "#FFFFFF"
    const val DARK_GRAY = "#1F2937"
    
    // Background Colors
    const val CARD_BACKGROUND = "#FFFFFF"
}

/**
 * Пример использования в Compose
 */
/*
@Composable
fun StormSkyExample() {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(StormSkyColors.stormSkyGradientColors[0])
            .padding(16.dp)
    ) {
        // Заголовок с золотым акцентом
        Text(
            text = "ALADDIN Security",
            style = MaterialTheme.typography.h3,
            color = StormSkyColors.goldMainCompose,
            modifier = Modifier.shadow(4.dp, RoundedCornerShape(8.dp))
        )
        
        // Карточка в стиле грозового неба
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(8.dp),
            backgroundColor = StormSkyColors.cardBackgroundCompose,
            elevation = 8.dp
        ) {
            Column(
                modifier = Modifier.padding(16.dp)
            ) {
                Text(
                    text = "Статус безопасности",
                    style = MaterialTheme.typography.h6,
                    color = StormSkyColors.darkGrayCompose
                )
                
                Text(
                    text = "Защищено",
                    style = MaterialTheme.typography.body1,
                    color = StormSkyColors.successGreenCompose
                )
            }
        }
        
        // Золотая кнопка
        Button(
            onClick = { /* Action */ },
            colors = ButtonDefaults.buttonColors(
                containerColor = StormSkyColors.goldMainCompose
            ),
            modifier = Modifier
                .fillMaxWidth()
                .padding(8.dp)
        ) {
            Text(
                text = "Подключить VPN",
                color = ComposeColor.White
            )
        }
    }
}
*/

